src/main/java/me/xmrvizzy/skyblocker/skyblock/api/ProfileUtils.java

```diff
@@ -0,0 +1,110 @@
+package me.xmrvizzy.skyblocker.skyblock.api;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import marcono1234.gson.recordadapter.RecordTypeAdapterFactory;
+import me.xmrvizzy.skyblocker.skyblock.api.records.PlayerProfiles;
+import me.xmrvizzy.skyblocker.skyblock.itemlist.ItemFixerUpper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.*;
+import net.minecraft.text.Text;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public class ProfileUtils {
+    public static PlayerProfiles getProfiles(String name){
+        try {
+            URL url = new URL("https://sky.shiiyu.moe/api/v2/profile/" + name);
+            InputStreamReader reader = new InputStreamReader(url.openStream());
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapterFactory(RecordTypeAdapterFactory.builder().allowMissingComponentValues().create())
+                    .serializeNulls()
+                    .create();
+            return gson.fromJson(reader, PlayerProfiles.class);
+        } catch (IOException e) {
```

Comment:
Since we're not doing anything special with the exceptions, maybe use `Exception` instead of `IOException`? Would catch all other unexpected things.